### PR TITLE
Fix workspace deletion e2e flake

### PR DESCRIPTION
**Pull request by qiujian16**
1. increase wait time
2. pass the test when list workspace return 404

## Related issue(s)

Fixes https://github.com/kcp-dev/kcp/issues/950

### CODE REVIEW

File: test/e2e/reconciler/clusterworkspacedeletion/controller_test.go [hunk 1]
```diff
@@ -227,19 +227,23 @@ func TestWorkspaceDeletionController(t *testing.T) {
 				t.Logf("Ensure workspace in the org workspace is deleted")
 				require.Eventually(t, func() bool {
 					wslist, err := server.orgKcpClient.TenancyV1alpha1().ClusterWorkspaces().List(ctx, metav1.ListOptions{})
+					// 404 could be returned if the org workspace is deleted.
```
Comment (stevekuznetsov) [inline test/e2e/reconciler/clusterworkspacedeletion/controller_test.go:230]: Why isn't this also true for the namespace list calls above this?

File: test/e2e/reconciler/clusterworkspacedeletion/controller_test.go [hunk 2]
```diff
@@ -227,19 +227,23 @@ func TestWorkspaceDeletionController(t *testing.T) {
...
+						return true
+					}
+
 					if err != nil {
 						return false
 					}
 
 					return len(wslist.Items) == 0
-				}, wait.ForeverTestTimeout, 1*time.Second)
+				}, 2*wait.ForeverTestTimeout, 1*time.Second)
```
Comment (stevekuznetsov) [inline test/e2e/reconciler/clusterworkspacedeletion/controller_test.go:240]: 2 * forever = smell. Do we know what takes more than 30s here?

Comment (ncdc) [reply test/e2e/reconciler/clusterworkspacedeletion/controller_test.go:240]: I actually am thinking that handling the 404 will fix the problem, so we could start with that, and not do this 2* change now

File: test/e2e/reconciler/clusterworkspacedeletion/controller_test.go (hunk 1 above)
Comment (ncdc) [reply test/e2e/reconciler/clusterworkspacedeletion/controller_test.go:230]: I believe this is coming from https://github.com/kcp-dev/kcp/blob/decced493792aeda1e171b9c860cc9efbcff00fd/pkg/server/apiextensions.go#L147. I see this getting logged repeatedly when the e2e is about to fail, presumably from the test trying to list clusterworkspaces every 1s

Comment (stevekuznetsov) [reply test/e2e/reconciler/clusterworkspacedeletion/controller_test.go:230]: Hm, so you don't get a 404 when you ask for a built-in type even when the workspace does not exist? Interesting.

**Review by stevekuznetsov — APPROVED** [review]

File: test/e2e/reconciler/clusterworkspacedeletion/controller_test.go (hunk 1 above)
Comment (ncdc) [reply test/e2e/reconciler/clusterworkspacedeletion/controller_test.go:230]:
```
k get --raw '/clusters/root:bob/api/v1/namespaces'
{"kind":"NamespaceList","apiVersion":"v1","metadata":{"resourceVersion":"1983"},"items":[]}
```

Comment (sttts) [reply test/e2e/reconciler/clusterworkspacedeletion/controller_test.go:230]: What I don't like here: with this change we do not test what we wanted to test anymore. I think we have to use this etcd client pattern of checking directly in etcd. A 404 of a list does not say anything about deletion of the objects.

Comment (ncdc) [reply test/e2e/reconciler/clusterworkspacedeletion/controller_test.go:230]: https://github.com/kcp-dev/kcp/issues/992

Comment (ncdc) [reply test/e2e/reconciler/clusterworkspacedeletion/controller_test.go:230]: Let's get this in, then follow up

**Comment by ncdc** [comment]
Rebased on top of main to pick up fix for #947

**Pull request merged**